File: go4java/Lambdas.go
package main

import "fmt"

// START_RUNNER OMIT
type Runner struct{}

func (r Runner) Run(f func()) { f() }

// START_TASK OMIT
type Task struct{ msg string }

func (t Task) Run() { fmt.Println("running", t.msg) }

// END_TASK OMIT

func main() {
	// RUN_LIT OMIT
	r := Runner{}
	r.Run(func() { fmt.Println("running a literal function") })

	// RUN_METHOD OMIT
	t := Task{"an method value"}
	r.Run(t.Run)
	// END_MAIN OMIT
}

File: go4java/Lambdas.java
class Lambdas {
    // START OMIT
    interface Task {
        void run();
    }

    class Runner {
        void run(Task t) {
            t.run();
        }
    }
    // END OMIT
    void test() {
        // ANONYMOUS OMIT
        Runner runner = new Runner();
        runner.run(new Task() {
            public void run() {
                System.out.println("an anonymous interface");
            };
        });
        // LAMBDA OMIT
        runner.run(() ->System.out.println("a lambda expression"));
    }

    public static void main(String[] args) {
        new Lambdas().test();
    }
}